routing/probability_estimator.go

```diff
@@ -81,19 +81,18 @@ func (p *probabilityEstimator) getNodeProbability(now time.Time,
 	totalWeight := aprioriFactor
 
 	for _, result := range results {
-		age := now.Sub(result.Timestamp)
-
-		switch {
-		// Weigh success with a constant high weight of 1. There is no
-		// decay.
-		case result.Success:
+		// Weigh success with a constant high weight of 1. There
+		// is no decay.
+		if result.SuccessAmt != 0 && amt <= result.SuccessAmt {
```

Comment:
Shouldn't `SuccessTIme` be used here,, similar to how `FailTIme.IsZero` is checked below?

Reply:
The condition can actually be removed. If there is no success recorded (or pushed down by an amt independent failure), `successAmt` will be zero and amt is always greater than zero.